code/sorting/src/insertion_sort/insertion_sort.cpp

```diff
@@ -1,49 +1,48 @@
-/*
- Part of Cosmos by OpenGenus Foundation
 
- insertion sort synopsis
-
-template<typename _Bidirectional_Iter, typename _Compare>
-void
-insertionSort(_Bidirectional_Iter begin, _Bidirectional_Iter end, _Compare compare);
-
-template<typename _Bidirectional_Iter>
-void
-insertionSort(_Bidirectional_Iter begin, _Bidirectional_Iter end);
- */
-
-#include <functional>
-
-template<typename _Bidirectional_Iter, typename _Compare>
-void
-insertionSort(_Bidirectional_Iter begin, _Bidirectional_Iter end, _Compare compare)
+// C program for insertion sort
+#include <stdio.h>
+#include <math.h>
+ 
+/* Function to sort an array using insertion sort*/
+void insertionSort(int arr[], int n)
```

Comment:
The insertionSort of the e333467 is a general version, I think it should not be removed.
You could find usage for it [here](https://github.com/OpenGenus/cosmos/blob/master/code/sorting/test/test_sort.cpp)